Author: Bernhard M. Wiedemann <bwiedemann suse de>
Date: 2018-07-23

https://savannah.nongnu.org/support/index.php?109535

Allow to override build date
in order to make builds reproducible.
See https://reproducible-builds.org/ for why this is good
and https://reproducible-builds.org/specs/source-date-epoch/
for the definition of this variable.
This date call only works with GNU date.

--- config/version.sh.orig
+++ config/version.sh
@@ -6,13 +6,13 @@
 set -e
 
 version=${1?}
-host=`uname -n`
+host=${HOSTNAME:-`uname -n`}
 if test -d "$srcdir/.git"; then
     git=" `git -C $srcdir describe --long --dirty`"
 else
     git=
 fi
-date="`TZ=GMT0 date +'%Y-%m-%d %T'` +0000"
+date="$(TZ=GMT0 date --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}" '+%Y-%m-%d %T +0000')"
 
 cat <<E
 char *version_str = "nmh-$version$git built $date on $host";
--- man/mh-chart-gen.sh.orig
+++ man/mh-chart-gen.sh
@@ -13,7 +13,7 @@ nmhmandir=$(dirname $0)
 # from the local build environment when building distribution packages.
 LC_TIME=C; export LC_TIME
 unset LANG
-datestamp=$(date +%Y-%m-%d)
+datestamp="$(date --utc --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}" '+%B %d, %Y')"
 
 our_name="mh-chart"
 our_desc="chart of all nmh commands and their options"
